### PR TITLE
PHRAS-442 #time 5h

### PR DESCRIPTION
**Pull request by jygaulier**
facets returned in API
"suggestions" returned in API (as flat facets)

### CODE REVIEW

File: lib/Alchemy/Phrasea/Controller/Api/V1Controller.php [hunk 1]
```diff
@@ -1081,6 +1081,7 @@ private function prepareSearchRequest(Request $request)
                 function (SearchEngineSuggestion $suggestion) {
                     return $suggestion->toArray();
                 }, $search_result->getSuggestions()->toArray()),
+            'facets'            => $search_result->getFacets(),
```
Comment (bburnichon) [inline lib/Alchemy/Phrasea/Controller/Api/V1Controller.php]: Can you please remove superfluous spaces? Thanks

File: lib/Alchemy/Phrasea/SearchEngine/Elastic/Search/FacetsResponse.php [hunk 2]
```diff
@@ -31,6 +31,11 @@ public function __construct(Escaper $escaper, array $response)
         }
     }
 
+    public function getFacets()
```
Comment (bburnichon) [inline lib/Alchemy/Phrasea/SearchEngine/Elastic/Search/FacetsResponse.php]: Please add phpdoc block to ease IDE in guessing types.

**Comment by bburnichon** [comment]
Seems you broke 2 tests.

File: lib/Alchemy/Phrasea/SearchEngine/Elastic/Search/FacetsResponse.php [hunk 3]
```diff
@@ -31,6 +31,16 @@ public function __construct(Escaper $escaper, array $response)
...
     }
 
+    /**
+     * Term aggregates
+     *
+     * @return array
+     */
+    public function getFacets()
+    {
+        return $this->facets;
```
Comment (mdarse) [inline lib/Alchemy/Phrasea/SearchEngine/Elastic/Search/FacetsResponse.php]: You should not do that. This is private structure, if you expose it we will not be able to change it anymore.

File: lib/Alchemy/Phrasea/SearchEngine/Elastic/ElasticSearchEngine.php [hunk 4]
```diff
@@ -292,8 +293,16 @@ public function query($string, $offset, $perPage, SearchEngineOptions $options =
...
         }
 
+        /** @var FacetsResponse $facets */
         $facets = $this->facetsResponseFactory->__invoke($res);
 
+        // for es, suggestions are a flat view of facets (api backward compatibility)
+        $suggestions->clear();
+        foreach($facets->getFacets() as $facet) {
+            foreach($facet['values'] as $value) {
+                $suggestions->add(new SearchEngineSuggestion($value['query'], $value['value'], $value['count']));
```
Comment (mdarse) [inline lib/Alchemy/Phrasea/SearchEngine/Elastic/ElasticSearchEngine.php]: This code should not be here, at least it should be extracted in another method or class, `ElasticSearchEngine` is too heavy already. Maybe this belong in some `FacetResponse::getSuggestions()` method.

**Pull request merged**